### DISCUSSION TITLE
Is there any way to do pagination with a start and end uuid?

### DISCUSSION THREAD

**Discussion by arichyx** [prisma/prisma#24940] [Q&A]
### Question
Is there any way to do pagination with a start and end **uuid**? Something just like
```typescript
prisma.posts.findMany({
  orderBy: { created_at: 'desc' },
  where: {
    id: { lte: startId, gte: endId },
  }
});
```
The problem is, the uuid is randomly generated and not sequential, I don't know how to do pagination with it.

### Background
I'm developing a demo proj just like twitter. There is a post feed, and a textarea to send new post. The post feed list is ordered by `created_at` desc.

What I'm trying to do is, when the user sends a new post, I want to fetch all posts between the new post and the first (current latest) post in the list, and insert them before the first post.

**Comment by ludralph** [comment]
Hi @Arichy 👋 

You can use a combination of the `created_at` timestamp and the `UUID`. This approach ensures that you can paginate through the posts effectively even if the UUIDs are not sequential. 

- Order by created_at and id: Since `created_at` might not be unique, you should also order by `id` to ensure consistent ordering.

- Use a cursor for pagination: You can use a cursor that combines both `created_at` and `id` to paginate through the posts.

For example:
```js
const posts = await prisma.posts.findMany({
  orderBy: [
    { created_at: 'desc' },
    { id: 'desc' }
  ],
  where: {
    created_at: {
      lte: startCreatedAt,
      gte: endCreatedAt
    },
    id: {
      lte: startId,
      gte: endId
    }
  },
  cursor: {
    id: startId
  },
  take: 10, // Adjust the number of posts you want to fetch
  skip: 1 // Skip the cursor itself
});
```

You can also see this closed [issue](https://github.com/prisma/prisma/issues/2855#issuecomment-673828649).

**Reply by arichyx** [reply]
Thanks for reply. But after test, `id` would be sorted by alphabet order, instead of the default order in DB.

**Reply by ludralph** [reply]
Hi @Arichy 

The behavior you've observed is indeed correct. Some databases offer UUID types that maintain insertion order. For example, PostgreSQL has extensions like uuid-ossp that provide functions to generate time-ordered UUIDs. However, this approach is database-specific and may not be available or consistent across all databases that Prisma supports.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
